packages/renderer/src/lib/task-manager/ProgressBar.svelte

```diff
@@ -29,12 +29,19 @@
 </style>
 
 <script lang="ts">
-export let progress: number | undefined;
-export let width: string = 'w-36';
-export let height: string = 'h-4';
+import type { HTMLAttributes } from 'svelte/elements';
+
+interface Props extends HTMLAttributes<HTMLElement> {
+  progress?: number;
+  width?: string;
+  height?: string;
+  class?: string;
+}
+
+let { progress, width = 'w-36', height = 'h-4', class: className, ...restProps }: Props = $props();
 </script>
 
-<div class="flex flex-row">
+<div class="flex flex-row {className}" {...restProps} >
```

Comment:
```suggestion
<div class="flex flex-row {restProps.class}" {...restProps} >
```

Reply:
This would duplicate the class attribute, leading to one being overwritten ?

Reply:
I'm wondering if it should not be {...restProps} first as I don't know it'll handle two `class` attribute (last win or first win ?)

Reply:
first win

Reply:
> This would duplicate the class attribute, leading to one being overwritten ?

I think it'll keep only the last one

Reply:
https://github.com/user-attachments/assets/2a8f6a6e-0d3e-4901-be84-bf69e94ba4ed

Reply:
`<div  {...restProps} class="flex flex-row {restProps.class}">` I don't know about your 'first' but svelte will keep only the class field `class="flex flex-row {restProps.class}"` even if defined in restProps in this order. So to me it's last definition = winner.

In this order `<div  class="flex flex-row {restProps.class}"  {...restProps}>` it will only keep class with the value from `restProps`

Reply:
This synthax is problematic, as a reader might not understand what is happening when trying to figure out how the component works.

Since this seems to rely on under the hood stuff that decide which one should be applied, I have a preference for the one where the class is not included in the restProps to avoid having a misunderstanding of how it operate :/

Reply:
I didn't use devtools console, I modified the source code and it's last win

Reply:
Yes I understand your point, and this is valid, but this is not something trivial, so I am not sure that this would be easy to understand, because this is not something we would document on each component